tests/Shared/Aspire.Workload.Testing.targets

```diff
@@ -169,4 +169,68 @@
+      <MissingPackages Include="@(ExpectedPackagePaths)" Condition="!Exists(%(ExpectedPackagePaths.Identity))" />
+
+      <UnexpectedPackages Include="@(AllPackages)" Exclude="@(ExpectedPackagePaths)" />
+
+      <!-- Ignore the packages which are part of the workload. Existence of these would get checked
+            by the workload installation targets. And their names don't correspond to project names. -->
+      <UnexpectedPackages Remove="@(UnexpectedPackages)" Condition="'%(UnexpectedPackages.FileName)' == 'Aspire.Hosting.Sdk.$(PackageVersion)'" />
+      <UnexpectedPackages Remove="@(UnexpectedPackages)" Condition="$([System.String]::Copy('%(UnexpectedPackages.FileName)').StartsWith('Microsoft.NET.Sdk.Aspire.Manifest-'))" />
+      <UnexpectedPackages Remove="@(UnexpectedPackages)" Condition="$([System.String]::Copy('%(UnexpectedPackages.FileName)').StartsWith('Aspire.Hosting.Orchestration.'))" />
+      <UnexpectedPackages Remove="@(UnexpectedPackages)" Condition="$([System.String]::Copy('%(UnexpectedPackages.FileName)').StartsWith('Aspire.Dashboard.Sdk.'))" />
+    </ItemGroup>
+
+    <Warning Text="Found some unexpected packages in '$(ArtifactsShippingPackagesDir)', which might mean that the list of expected packages is incorrect: @(UnexpectedPackages -> '%(FileName)%(Extension)', ', '). Use %24(SkipPackageCheckForWorkloadTesting)='true' to skip this."
+             Condition="@(UnexpectedPackages -> Count()) &gt; 0" />
+    <Error Text="Missing some expected packages in '$(ArtifactsShippingPackagesDir)': @(MissingPackages -> '%(FileName)%(Extension)',', ') . Make sure to build with `-pack`. Use %24(SkipPackageCheckForWorkloadTesting)='true' to skip this."
+           Condition="@(MissingPackages -> Count()) &gt; 0" />
+  </Target>
+
+  <UsingTask TaskName="GetExpectedNuGetPackageNames"
+      TaskFactory="RoslynCodeTaskFactory"
+      AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <SourceDirectory ParameterType="System.String" Required="true" />
+      <PackagesDirectory ParameterType="System.String" Required="true" />
+      <PackageNames ParameterType="System.String[]" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+            if (!Directory.Exists(SourceDirectory))
+            {
+                Log.LogError($"Could not find {nameof(SourceDirectory)}={SourceDirectory}");
+                return false;
+            }
+            if (!Directory.Exists(PackagesDirectory))
+            {
+                Log.LogError($"Could not find {nameof(PackagesDirectory)}={PackagesDirectory}");
+                return false;
+            }
+
+            var files = Directory.EnumerateFiles(SourceDirectory, "*.csproj", SearchOption.AllDirectories);
+            PackageNames = files.Where(f => File.ReadAllText(f).Contains("<IsPackable>true</IsPackable>"))
```

Comment:
This would break if we move the IsPackable to a Directory.Build.props file, right? I wonder if instead of using an inline task if it's best to just use MSBuild task for all of these projects and call a target that just returns the value of IsPackable. We don't have to do it that way but I just think that would probably be more future proof and resilient against refactorings.

Reply:
Thanks for the suggestion! Changed.